Create CI troubleshooting guide with failure patterns and fix strategies

## Task Description
Create `docs/CI_TROUBLESHOOTING.md` documenting all known CI failure patterns, their root causes, and fix strategies for this project.

## CI Pipeline Overview
This project runs CI on every PR and push to main/dev. See `.github/workflows/ci.yml` header for pass criteria.

**Pass Criteria (all run in `dashboard/` directory):**
1. `npm run build` (clean React 19 production build)
2. `npm run lint` (zero ESLint errors)

## Known Failure Patterns to Document

### 1. Build Failures from Bad Imports
- **Symptom**: `Module not found` or `Failed to resolve import` during build
- **Root Cause**: Importing a component or module that does not exist
- **Troubleshooting**: Check the import path. Verify the target file exists in `dashboard/src/`. Run `npm run build` in `dashboard/` locally.
- **Prevention**: Always verify import targets exist before committing.

### 2. ESLint Errors
- **Symptom**: `npm run lint` exits with errors
- **Root Cause**: Code style violations, unused variables, missing types
- **Troubleshooting**: Run `cd dashboard && npm run lint` locally. Fix each error. Common fixes: remove unused imports, add missing return types.
- **Prevention**: Run lint before committing.

### 3. React 19 Compatibility Issues
- **Symptom**: Build warnings or errors about deprecated React APIs
- **Root Cause**: Using React 18 patterns that changed in React 19
- **Troubleshooting**: Check React 19 migration guide. Common changes: `ReactDOM.render` replaced, new JSX transform.
- **Prevention**: Use functional components with hooks. Avoid deprecated lifecycle methods.

### 4. Working Directory Mistakes
- **Symptom**: `npm ERR! missing script` or `package.json not found`
- **Root Cause**: Running commands in repo root instead of `dashboard/` subdirectory
- **Troubleshooting**: All npm commands must run inside `dashboard/`. Check `working-directory: dashboard` in CI.
- **Prevention**: Always `cd dashboard` before running npm commands.

### 5. Static Site vs Dashboard Confusion
- **Symptom**: Changes to static site files break dashboard build or vice versa
- **Root Cause**: This repo has two parts -- a static site (root) and a React dashboard (`dashboard/`). CI only checks the dashboard.
- **Troubleshooting**: Identify which part your changes affect. Dashboard changes need lint+build. Static site changes are not CI-checked.

## Optimal Troubleshooting Strategy
1. **Check which job failed** -- build or lint? Each has different fixes
2. **Reproduce locally** -- `cd dashboard && npm run build && npm run lint`
3. **Read the error line** -- CI logs show exact file and line number
4. **Fix one error at a time** -- lint errors cascade, fix first one and re-run
5. **Verify working directory** -- all commands run in `dashboard/`, not repo root

## Requirements
- [ ] Create `docs/CI_TROUBLESHOOTING.md` with all patterns above
- [ ] Add any additional failure patterns found by examining the workflow and recent CI runs
- [ ] Include the troubleshooting strategy
- [ ] Keep it concise and actionable

## Acceptance Criteria
- [ ] Document exists at `docs/CI_TROUBLESHOOTING.md`
- [ ] All failure patterns documented with symptom, root cause, troubleshooting, prevention
- [ ] All CI checks pass

## Files to Modify/Create
- `docs/CI_TROUBLESHOOTING.md` (create)

## Technical Constraints
- Markdown only, no code changes
- Follow .kiro/steering/ standards (no emojis)
---
**Note**: Worker will execute with `--trust-all-tools`.

--- a/.github/workflows/ci-fix-dispatcher.yml
+++ b/.github/workflows/ci-fix-dispatcher.yml
@@ -74,41 +74,6 @@ jobs:
           gh issue create --repo "$REPO" \
             --title "CI fix: $BRANCH ($CATEGORY)" \
             --label "kiro,queued" \
-            --body "## Task Description
-Fix the CI failure on branch \`$BRANCH\` (original issue #$ISSUE_NUM).
-
-## Error Category
-**$CATEGORY**
-
-## Fix Instructions
-$FIX_HINT
-
-## CI Failure Logs
-\`\`\`
-$LOGS_TRUNCATED
-\`\`\`
-
-## Requirements
-- [ ] Fix the error shown in the logs above
-- [ ] Ensure all CI checks pass: lint, type-check, test, build
-- [ ] Do NOT change test thresholds or disable lint rules
-
-## Acceptance Criteria
-- [ ] CI passes on the branch
-- [ ] No new lint warnings introduced
-
-## Files to Modify/Create
-- Check branch \`$BRANCH\` for the files that need fixing
-
-## Technical Constraints
-- Work on branch \`$BRANCH\` (checkout existing branch, do not create new)
-- Only fix the CI errors, do not add new features
-- Follow .kiro/steering/ standards
-
-## Context
-CI run: https://github.com/$REPO/actions/runs/$RUN_ID
-Original issue: #$ISSUE_NUM
----
-**Note**: Worker will execute with \`--trust-all-tools\`."
+            --body "## Task Description\nFix the CI failure on branch \`$BRANCH\` (original issue #$ISSUE_NUM).\n\n## Error Category\n**$CATEGORY**\n\n## Fix Instructions\n$FIX_HINT\n\n## CI Failure Logs\n\`\`\`\n$LOGS_TRUNCATED\n\`\`\`\n\n## Requirements\n- [ ] Fix the error shown in the logs above\n- [ ] Ensure all CI checks pass: lint, type-check, test, build\n- [ ] Do NOT change test thresholds or disable lint rules\n\n## Acceptance Criteria\n- [ ] CI passes on the branch\n- [ ] No new lint warnings introduced\n\n## Files to Modify/Create\n- Check branch \`$BRANCH\` for the files that need fixing\n\n## Technical Constraints\n- Work on branch \`$BRANCH\` (checkout existing branch, do not create new)\n- Only fix the CI errors, do not add new features\n- Follow .kiro/steering/ standards\n\n## Context\nCI run: https://github.com/$REPO/actions/runs/$RUN_ID\nOriginal issue: #$ISSUE_NUM\n---\n**Note**: Worker will execute with \`--trust-all-tools\`."
 
           echo "Fix issue created for $BRANCH ($CATEGORY)"

--- a/.github/workflows/ci-fix-dispatcher.yml
+++ b/.github/workflows/ci-fix-dispatcher.yml
@@ -71,9 +71,46 @@ jobs:
           LOGS_TRUNCATED=$(echo "$LOGS" | tail -30)
 
           # Create fix issue
+          ISSUE_BODY="## Task Description
+Fix the CI failure on branch \`$BRANCH\` (original issue #$ISSUE_NUM).
+
+## Error Category
+**$CATEGORY**
+
+## Fix Instructions
+$FIX_HINT
+
+## CI Failure Logs
+\`\`\`
+$LOGS_TRUNCATED
+\`\`\`
+
+## Requirements
+- [ ] Fix the error shown in the logs above
+- [ ] Ensure all CI checks pass: lint, type-check, test, build
+- [ ] Do NOT change test thresholds or disable lint rules
+
+## Acceptance Criteria
+- [ ] CI passes on the branch
+- [ ] No new lint warnings introduced
+
+## Files to Modify/Create
+- Check branch \`$BRANCH\` for the files that need fixing
+
+## Technical Constraints
+- Work on branch \`$BRANCH\` (checkout existing branch, do not create new)
+- Only fix the CI errors, do not add new features
+- Follow .kiro/steering/ standards
+
+## Context
+CI run: https://github.com/$REPO/actions/runs/$RUN_ID
+Original issue: #$ISSUE_NUM
+---
+**Note**: Worker will execute with \`--trust-all-tools\`."
+
           gh issue create --repo "$REPO" \
             --title "CI fix: $BRANCH ($CATEGORY)" \
             --label "kiro,queued" \
-            --body "## Task Description\nFix the CI failure on branch \`$BRANCH\` (original issue #$ISSUE_NUM).\n\n## Error Category\n**$CATEGORY**\n\n## Fix Instructions\n$FIX_HINT\n\n## CI Failure Logs\n\`\`\`\n$LOGS_TRUNCATED\n\`\`\`\n\n## Requirements\n- [ ] Fix the error shown in the logs above\n- [ ] Ensure all CI checks pass: lint, type-check, test, build\n- [ ] Do NOT change test thresholds or disable lint rules\n\n## Acceptance Criteria\n- [ ] CI passes on the branch\n- [ ] No new lint warnings introduced\n\n## Files to Modify/Create\n- Check branch \`$BRANCH\` for the files that need fixing\n\n## Technical Constraints\n- Work on branch \`$BRANCH\` (checkout existing branch, do not create new)\n- Only fix the CI errors, do not add new features\n- Follow .kiro/steering/ standards\n\n## Context\nCI run: https://github.com/$REPO/actions/runs/$RUN_ID\nOriginal issue: #$ISSUE_NUM\n---\n**Note**: Worker will execute with \`--trust-all-tools\`."
+            --body "$ISSUE_BODY"
 
           echo "Fix issue created for $BRANCH ($CATEGORY)"

--- a/docs/CI_TROUBLESHOOTING.md
+++ b/docs/CI_TROUBLESHOOTING.md
@@ -1,0 +1,148 @@
+# CI Troubleshooting Guide
+
+This document covers common CI failure patterns, their root causes, and fix strategies for this project.
+
+## CI Pipeline Overview
+
+The CI pipeline runs on every PR and push to main/dev branches. It consists of two jobs:
+
+1. **root-build**: Builds the static site in the repository root
+2. **dashboard-build**: Builds and lints the React dashboard in the `dashboard/` directory
+
+Both jobs must pass for CI to succeed.
+
+### Dashboard Pass Criteria
+
+All commands run in the `dashboard/` directory:
+
+1. `npm run build` - Clean React 19 production build with no errors
+2. `npm run lint` - Zero ESLint errors
+
+## Known Failure Patterns
+
+### 1. Build Failures from Bad Imports
+
+**Symptom**: `Module not found` or `Failed to resolve import` during build
+
+**Root Cause**: Importing a component or module that does not exist or has an incorrect path
+
+**Troubleshooting**:
+- Check the import path in the error message
+- Verify the target file exists in `dashboard/src/`
+- Run `cd dashboard && npm run build` locally to reproduce
+- Check for typos in file names or paths
+- Verify the file extension matches (.tsx, .ts, .jsx, .js)
+
+**Prevention**: Always verify import targets exist before committing
+
+### 2. ESLint Errors
+
+**Symptom**: `npm run lint` exits with errors
+
+**Root Cause**: Code style violations, unused variables, missing types, or other linting rule violations
+
+**Troubleshooting**:
+- Run `cd dashboard && npm run lint` locally
+- Fix each error reported by ESLint
+- Common fixes:
+  - Remove unused imports and variables
+  - Add missing return types
+  - Fix indentation and formatting
+  - Add missing dependencies to useEffect hooks
+
+**Prevention**: Run lint before committing changes
+
+### 3. React 19 Compatibility Issues
+
+**Symptom**: Build warnings or errors about deprecated React APIs
+
+**Root Cause**: Using React 18 patterns that changed or were removed in React 19
+
+**Troubleshooting**:
+- Check React 19 migration guide for breaking changes
+- Common changes:
+  - `ReactDOM.render` replaced with `createRoot`
+  - New JSX transform (no need to import React in every file)
+  - Changes to ref handling and forwarding
+
+**Prevention**: Use functional components with hooks. Avoid deprecated lifecycle methods.
+
+### 4. Working Directory Mistakes
+
+**Symptom**: `npm ERR! missing script` or `package.json not found`
+
+**Root Cause**: Running commands in the repository root instead of the `dashboard/` subdirectory
+
+**Troubleshooting**:
+- All dashboard npm commands must run inside `dashboard/`
+- Check that you're in the correct directory: `pwd` should show `.../dashboard`
+- The CI workflow uses `working-directory: dashboard` for all dashboard commands
+
+**Prevention**: Always `cd dashboard` before running npm commands for the dashboard
+
+### 5. Static Site vs Dashboard Confusion
+
+**Symptom**: Changes to static site files break dashboard build or vice versa
+
+**Root Cause**: This repository has two parts:
+- Static site (repository root)
+- React dashboard (`dashboard/` subdirectory)
+
+CI checks both, but they are independent projects with separate dependencies.
+
+**Troubleshooting**:
+- Identify which part your changes affect
+- Dashboard changes require passing both `npm run build` and `npm run lint` in `dashboard/`
+- Static site changes require passing `npm run build` in the repository root
+- Each has its own `package.json` and dependencies
+
+**Prevention**: Understand which part of the project you're modifying
+
+### 6. Dependency Issues
+
+**Symptom**: Build fails with missing package errors or version conflicts
+
+**Root Cause**: `package-lock.json` out of sync with `package.json`, or missing dependencies
+
+**Troubleshooting**:
+- Run `cd dashboard && npm ci` to install exact versions from lock file
+- If adding new dependencies, ensure `package-lock.json` is committed
+- Check that the dependency exists in `package.json`
+
+**Prevention**: Always commit both `package.json` and `package-lock.json` when adding dependencies
+
+## Optimal Troubleshooting Strategy
+
+1. **Check which job failed** - Look at the CI logs to see if it was root-build or dashboard-build, and whether it failed on build or lint
+2. **Reproduce locally** - Run the exact commands that failed:
+   - For dashboard: `cd dashboard && npm run build && npm run lint`
+   - For root: `npm run build`
+3. **Read the error line** - CI logs show the exact file and line number where the error occurred
+4. **Fix one error at a time** - Lint errors can cascade; fix the first error and re-run to see if others resolve
+5. **Verify working directory** - Ensure you're running commands in the correct directory (dashboard commands in `dashboard/`, root commands in root)
+6. **Test before pushing** - Run both build and lint locally before pushing to ensure CI will pass
+
+## Quick Reference
+
+### Local Testing Commands
+
+```bash
+# Test dashboard build and lint
+cd dashboard
+npm ci
+npm run build
+npm run lint
+
+# Test root build
+cd ..
+npm ci
+npm run build
+```
+
+### Common Fixes
+
+- **Unused import**: Remove the import statement
+- **Missing type**: Add TypeScript type annotation
+- **Module not found**: Check import path and file existence
+- **Lint error**: Run `npm run lint` and follow the error messages
+- **Build error**: Run `npm run build` and check the stack trace

